src/Microsoft.AspNet.SignalR.Core/Hubs/TypedClientBuilder.cs

```diff
@@ -123,6 +123,11 @@ private static void BuildMethod(TypeBuilder type, MethodInfo interfaceMethodInfo
 
             methodBuilder.SetReturnType(interfaceMethodInfo.ReturnType);
             methodBuilder.SetParameters(paramTypes);
+            // Sets the number of generic type parameters
+            foreach (var generic in paramTypes.Where(param => param.IsGenericParameter))
+            {
+                methodBuilder.DefineGenericParameters(generic.Name);
```

Comment:
This won't work if the method takes multiple parameters. As per MSDN - . "Calling this method a second time causes an InvalidOperationException.". DefineGenericParameters takes param so the code will need to be changed to pass all the types at once.